File: Solved by Python/A. Train and Peter.py
# Nahin the Coder
# Date : 27 - 07 - 2021

s1 = input()
s2 = input()
s3 = input()

forward = bool(0)
backward = bool(0)

if s1.find(s2) != -1:
    if s1.find(s3, s1.find(s2) + len(s2)) != -1:
        forward = 1

bb = s1[::-1]

if bb.find(s2) != -1:
    if bb.find(s3, bb.find(s2) + len(s2)) != -1:
        backward = 1
if forward == 1 and backward == 1:
    print("both")
    exit()
if forward == 1:
    print("forward")
    exit()
if backward == 1:
    print("backward")
    exit()
print("fantasy")

File: Part 2/A - Train and Peter.cpp
//Solver : Nahin the Coder
//Date   : 27 - 07 - 2021
#include<bits/stdc++.h>
using namespace std;

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(0);
	
	bool forward = false;
	bool backward = false;
	bool both = false;

	
	
	string s1,s2,s3;
	cin>>s1>>s2>>s3;
	
	//forward check
	if(s1.find(s2) != -1)
	{
		if(s1.find(s3,s1.find(s2)+s2.length()) != -1)
		{
			forward = true;
		}
	}
	
	//backward check
	reverse(s1.begin(),s1.end());
	
	if(s1.find(s2) != -1)
	{
		if(s1.find(s3,s1.find(s2)+s2.length()) != -1)
		{
			backward = true;
		}
	}
	
	if(forward == true and backward == true)
	{
		cout<<"both";
		return 0;
	}
	if(forward == true)
	{
		cout<<"forward";
		return 0;
	}
	if(backward == true)
	{
		cout<<"backward";
		return 0;
	}
	cout<<"fantasy";
	
	return 0;
}
